enhancements/olm/catalogd-console-api.md

```diff
@@ -0,0 +1,670 @@
+                  
+Clients are also encouraged to implement local caching for frequently queried metadata.
+
+### Resource Considerations
+
+This proposal should have no impact on 
+SNO [resource minimums](https://docs.openshift.com/container-platform/4.17/installing/installing_sno/install-sno-preparing-to-install-sno.html#install-sno-requirements-for-installing-on-a-single-node_install-sno-preparing)
+- vCPU: 8
+- RAM: 16GiB
+- storage: 120GiB
+
+or Hypershift [SLOs](https://hypershift-docs.netlify.app/reference/slos/) where the primary consideration is memory uses in the 10s of MiB, not 100s.
+
+### Risks and Mitigations
+
+Depending on the implementation of this proposal, exercise of this endpoint could function as an 
+I/O multiplier which could starve other clients which are dependent on the same API. 
+The first iteration of this endpoint will not attempt to mitigate denial-of-service attempts via 
+rate-limiting or other resource consumption constraints.
+
+The proposal does not include any provision for authentication/authorization, since there are no 
+current use-cases which require it. 
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+-->
+
+### Drawbacks
+
+#### Caching 
+Given the highly-variable nature of responses based on unbound combinations of schema/name/package, 
+any server-side caching will be problematic if not completely impractical. Read: computationally and/or 
+storage expensive.
+
+For example, response for `metas?schema=olm.package&name=foo` and `metas?schema=olm.package&name=bar`
+represent disjoint sets, and the server would have to either ignore `If-Modified-Since` directives 
+in client requests or rely on their veracity uncritically.
+
+Further, if the underlying implementation relies on anything other than a dictionary of cached requests 
+(&parameters) to dictate date-relevant responses (for e.g. relying on the catalog unpack time), 
+then any server-side caching would be erroneous.  
+
+The existing `all` endpoint also incentivizes clients to conserve resources via local cache to avoid making 
+many (potentially duplicate) requests.  However, the OCP console proof of concept 
+required what was deemed an unsupportable amount of code and complexity to cache, decompose, and render the 
+complete FBC.
+
+#### Complexity
+Server implementation is much more complex and resource intensive due to the need to index catalog 
+content and serve variable requests.
+
+#### Completeness
+The previous `all` endpoint always returns valid FBC.  The new service cannot make that promise, 
+so clients could make incorrect assumptions about the suitability of results.  See Open Questions.
```

Comment:
Does a collection of valid FBC blobs not constitute `valid FBC`?

Reply:
Not necessarily.  Valid FBC blobs can include `metas?schema=olm.channel&name=foo` but this will not include bundles or packages, and would fail an `opm validate` call.   I'll add some lang that we're contrasting "valid FBC element collection" with "well-formed FBC for a package and constituents" .